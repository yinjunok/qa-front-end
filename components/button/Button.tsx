import * as React from 'react';
import * as css from './styles.less';

interface IButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  type?: 'primary' | 'blue' | 'grey' | 'vote';
}

const Button: React.SFC<IButtonProps> = ({children, className, type, ...props}) => {
  return (
    <button
      className={`${css.button} ${css[type as string]} ${className}`}
      {...props}
    >
      {children}
    </button>
  );
}

Button.defaultProps = {
  type: 'primary',
  className: ''
}

export default Button;
