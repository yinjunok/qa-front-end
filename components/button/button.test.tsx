import renderer from 'react-test-renderer';
import Button from './Button';

describe('test snapshots', () => {
  test('primary button', () => {
    const btn = renderer.create(
      <Button type='primary'>按钮</Button>
    );
    expect(btn.toJSON()).toMatchSnapshot();
  });

  test('blue button', () => {
    const btn = renderer.create(
      <Button type='blue'>按钮</Button>
    );
    expect(btn.toJSON()).toMatchSnapshot();
  });

  test('grey button', () => {
    const btn = renderer.create(
      <Button type='grey'>按钮</Button>
    );
    expect(btn.toJSON()).toMatchSnapshot();
  });

  test('vote button', () => {
    const btn = renderer.create(
      <Button type='vote'>按钮</Button>
    );
    expect(btn.toJSON()).toMatchSnapshot();
  });
});

test('extra className', () => {
  const btn = renderer.create(
    <Button type='vote'>按钮</Button>
  );
  expect(btn.toJSON()).toMatchSnapshot();
});