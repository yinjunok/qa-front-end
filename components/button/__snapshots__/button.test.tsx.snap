// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extra className 1`] = `
<button
  className="button vote "
>
  按钮
</button>
`;

exports[`test snapshots blue button 1`] = `
<button
  className="button blue "
>
  按钮
</button>
`;

exports[`test snapshots grey button 1`] = `
<button
  className="button grey "
>
  按钮
</button>
`;

exports[`test snapshots primary button 1`] = `
<button
  className="button primary "
>
  按钮
</button>
`;

exports[`test snapshots vote button 1`] = `
<button
  className="button vote "
>
  按钮
</button>
`;
